packages/wrangler/src/d1/execute.tsx

```diff
@@ -112,72 +116,79 @@ export async function executeSql(
 	}
 
 	return local
-		? await executeLocally(config, name, shouldPrompt, queries, persistTo)
-		: await executeRemotely(config, name, shouldPrompt, batchSplit(queries));
+		? await executeLocally(config, name, shouldPrompt, queries, persistTo, json)
+		: await executeRemotely(
+				config,
+				name,
+				shouldPrompt,
+				batchSplit(queries),
+				json
+		  );
 }
 type HandlerOptions = StrictYargsOptionsToInterface<typeof Options>;
-export const Handler = withConfig<HandlerOptions>(
-	async ({
+
+export const Handler = async (args: HandlerOptions): Promise<void> => {
+	const { local, database, yes, persistTo, file, command, json } = args;
+	if (json) {
+		// set loggerLevel to none to avoid readConfig warnings appearing in JSON output
+		logger.loggerLevel = "none";
```

Comment:
Do we need to capture the previous logging level and reset it, just in case this handler is used in a long lived process?

Reply:
Also, perhaps level "error" is actually needed here?
Otherwise things like the `logger.error()` on line 139 (and potentially other D1 errors) will be lost?
I think that `logger.error()` will be writing to `stderr` anyway, so it should not end up in the piped output for json.